mscore/editstyle.h

```diff
@@ -75,6 +75,8 @@ class EditStyle : public QDialog, private Ui::EditStyleBase {
 
       static const std::map<ElementType, EditStylePage> PAGES;
 
+      friend class MuseScore;
```

Comment:
??? "If you need to add friend classes, there are issues in the design of the codebase" (c)

Is it really needed?

Reply:
I had to make it friend in order to access ```EditStlye::stackPage``` and ```EditStyle::PageHeaderFooter``` from ```MuseScore::cmd```

> updated musescore.cpp line:6322
> _styleDlg->setPage(_styleDlg->pageStack->indexOf(_styleDlg->PageHeaderFooter));

There is one more option I can see is to change the inheritance of ```Ui::EditStyleBase``` to ```public``` in ```EditStyle```. But making class friend seems better option to me.

Let me know if there is any other way do access those method/elements without violating any design.